synnax/pkg/user/retrieve.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2024 Synnax Labs, Inc.
+//
+// Use of this software is governed by the Business Source License included in the file
+// licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with the Business Source
+// License, use of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt.
+
+package user
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/synnaxlabs/x/gorp"
+)
+
+// A Retrieve is used to retrieve users from the key-value store.
+type Retrieve struct {
+	// baseTX is the transaction that the Retrieve will use to atomically interact with
+	// the key-value store.
+	baseTX gorp.Tx
+	// gorp is the underlying query that the Retrieve will use to get users from the
+	// key-value store.
+	gorp gorp.Retrieve[uuid.UUID, User]
+}
+
+// WhereKeys filters the query to only include users with the given keys.
+func (r Retrieve) WhereKeys(keys ...uuid.UUID) Retrieve {
+	r.gorp = r.gorp.WhereKeys(keys...)
+	return r
+}
+
+// Entry binds the query to the given user.
+func (r Retrieve) Entry(user *User) Retrieve {
+	r.gorp = r.gorp.Entry(user)
+	return r
+}
+
+// Entries binds the query to the given users.
+func (r Retrieve) Entries(users *[]User) Retrieve {
+	r.gorp = r.gorp.Entries(users)
+	return r
+}
+
+// WhereUsername filters the query to only include users with the given username.
+func (r Retrieve) WhereUsername(username string) Retrieve {
```

Comment:
This function is unnecessary as `WhereUsernames` already takes a variadic parameter.